crypto/jwsutil/signature.go

```diff
@@ -0,0 +1,55 @@
+// Package jwsutil provides serialization utilities for JWT libraries.
+package jwsutil
+
+import (
+	"encoding/json"
+	"strings"
+)
+
+// Signature represents a detached signature.
+type Signature struct {
+	Protected   string          `json:"protected,omitempty"`
+	Unprotected json.RawMessage `json:"header,omitempty"`
```

Comment:
why the serialized name and property name are different?

Reply:
It is defined in [RFC 7515 7.2.1](https://datatracker.ietf.org/doc/html/rfc7515#section-7.2.1).